### PR TITLE
108269 Endpoints for creating Label and getting all Labels

### PR DESCRIPTION
**Pull request by janid1967**
[AB#108269](https://statoildeveloper.visualstudio.com/b6d97460-2e53-47ae-aad4-b41396d90828/_workitems/edit/108269)

### CODE REVIEW

File: src/Equinor.ProCoSys.Completion.Query/LabelQueries/GetAllLabels/LabelDto.cs [hunk 1]
```diff
@@ -0,0 +1,5 @@
+﻿using System.Collections.Generic;
+
+namespace Equinor.ProCoSys.Completion.Query.LabelQueries.GetAllLabels;
+
+public record LabelDto(string Text, bool IsVoided, List<string> EntitiesWithLabel);
```
Comment (jsoi-eq) [inline src/Equinor.ProCoSys.Completion.Query/LabelQueries/GetAllLabels/LabelDto.cs:5]: No LabelDtoValidator?

File: src/Equinor.ProCoSys.Completion.Command/LabelCommands/CreateLabel/CreateLabelCommandHandler.cs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        _unitOfWork = unitOfWork;
+        _logger = logger;
+    }
+
+    public async Task<Result<string>> Handle(CreateLabelCommand request, CancellationToken cancellationToken)
+    {
+        var label = new Label(request.Label);
+        _labelRepository.Add(label);
+
+        await _unitOfWork.SaveChangesAsync(cancellationToken);
```
Comment (jsoi-eq) [inline src/Equinor.ProCoSys.Completion.Command/LabelCommands/CreateLabel/CreateLabelCommandHandler.cs:30]: Labels are going to be global / plant independent?

File: src/Equinor.ProCoSys.Completion.Query/LabelQueries/GetAllLabels/LabelDto.cs (hunk 1 above)
Comment (janid1967) [reply src/Equinor.ProCoSys.Completion.Query/LabelQueries/GetAllLabels/LabelDto.cs:5]: No, this is a dto for a Query, delivered to client
xxxDtoValidators are Dtos received from client via endpoints

**Review by jsoi-eq — COMMENTED** [review]
Awaiting feedback.

File: src/Equinor.ProCoSys.Completion.WebApi/Controllers/Labels/CreateLabelDtoValidator.cs [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+public class CreateLabelDtoValidator : AbstractValidator<CreateLabelDto>
+{
+    public CreateLabelDtoValidator()
+    {
+        RuleLevelCascadeMode = CascadeMode.Stop;
+        ClassLevelCascadeMode = CascadeMode.Stop;
+
+        RuleFor(dto => dto).NotNull();
+
+        RuleFor(dto => dto.Text)
```
Comment (jsoi-eq) [inline src/Equinor.ProCoSys.Completion.WebApi/Controllers/Labels/CreateLabelDtoValidator.cs:14]: This validation allows empty string I believe, is that by design?

File: src/tests/Equinor.ProCoSys.Completion.Query.Tests/LabelQueries/GetAllLabels/GetAllLabelsQueryHandlerTests.cs [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+    //private Label _labelA;
+    //private Label _labelB;
+    //private Label _labelC;
+    //private Label _labelD;
+    //private Label _labelVoided;
+    private readonly GetAllLabelsQuery _query = new ();
+
+    protected override void SetupNewDatabase(DbContextOptions<CompletionContext> dbContextOptions)
+    {
+        //using var context = new CompletionContext(dbContextOptions, _plantProviderMockObject, _eventDispatcherMockObject, _currentUserProviderMockObject);
```
Comment (jsoi-eq) [inline src/tests/Equinor.ProCoSys.Completion.Query.Tests/LabelQueries/GetAllLabels/GetAllLabelsQueryHandlerTests.cs:26]: Lot of commented out code?

**Pull request merged**